### PR TITLE
Feature/bootstrap tenant

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

File: docs/docs/source/rest/rpc.rst [hunk 1]
```diff
@@ -24,8 +24,67 @@ RPC
...
+.. _rpc-bootstrap:
+
+Bootstrap Tenant
+================
+Bootstrapping a tenant will copy all providers, hardware types, image types, services, 
+cluster templates, and plugin resources from the superadmin to the tenant making the request.
+The user making the request must be the tenant admin. This is a copy, meaning if the 
+superadmin updates the entities, the changes will not be reflected in the tenant.
+
+To bootstrap a tenant, making a POST HTTP request to URI:
```
Comment (dereklwood) [inline docs/docs/source/rest/rpc.rst:38]: no URI... incomplete?

File: docs/docs/source/rest/rpc.rst [hunk 2]
```diff
@@ -24,8 +24,67 @@ RPC
...
+::
+
+ /bootstrap
+
+The body is a JSON Object containing one optional key ``overwrite`` which indicates whether or 
+not data from the superadmin should overwrite existing data. By default it is false, meaning an entity
+from the superadmin will not skipped if it already exists in the tenant account, even if the entities
```
Comment (dereklwood) [inline docs/docs/source/rest/rpc.rst]: ...will _be_ skipped... (doublecheck)

Comment (albertshau) [inline docs/docs/source/rest/rpc.rst]: yep. fingers are auto-completing poorly.

File: docs/docs/source/rest/rpc.rst (hunk 1 above)
Comment (albertshau) [inline docs/docs/source/rest/rpc.rst:38]: the path is just /v1/bootstrap which is below

File: docs/docs/source/rest/rpc.rst [hunk 3]
```diff
@@ -24,8 +24,67 @@ RPC
...
+================
+Bootstrapping a tenant will copy all providers, hardware types, image types, services, 
+cluster templates, and plugin resources from the superadmin to the tenant making the request.
+The user making the request must be the tenant admin. This is a copy, meaning if the 
+superadmin updates the entities, the changes will not be reflected in the tenant.
+
+To bootstrap a tenant, making a POST HTTP request to URI:
+::
+
+ /bootstrap
```
Comment (poornachandra) [inline docs/docs/source/rest/rpc.rst:41]: Can /bootstrap be called on a tenant multiple times? If so we should change the name to something else, right?

File: server/src/main/java/com/continuuity/loom/http/handler/LoomRPCHandler.java [hunk 4]
```diff
@@ -54,18 +61,98 @@
...
+   * @param request Request to bootstrap the tenant
+   * @param responder Responder for responding to the request
+   * @throws Exception
+   */
+  @POST
+  @Path("/bootstrap")
+  public void bootstrapTenant(HttpRequest request, HttpResponder responder) throws Exception {
+    Account account = getAndAuthenticateAccount(request, responder);
+    if (account == null) {
+      return;
```
Comment (poornachandra) [inline server/src/main/java/com/continuuity/loom/http/handler/LoomRPCHandler.java:103]: Is is possible to get a request with empty account?

**Comment by poornachandra** [comment]
Minor comments, LGTM :+1:

File: server/src/main/java/com/continuuity/loom/http/handler/LoomRPCHandler.java (hunk 4 above)
Comment (albertshau) [inline server/src/main/java/com/continuuity/loom/http/handler/LoomRPCHandler.java:103]: It's possible headers are missing, which is checked in getAndAuthenticateAccount.  The null check is really a check to see if that function found something wrong.

File: docs/docs/source/rest/rpc.rst (hunk 3 above)
Comment (albertshau) [inline docs/docs/source/rest/rpc.rst:41]: true.  will change this to only allow bootstrap if everything is completely empty.  If we want to support tenants using superadmin entities and having those entities update as the superadmin updates them, it should be something separate.  Maybe adding links or inheritance or something along those lines.

**Pull request merged**